diff --git a/container/bitnami/Dockerfile b/container/bitnami/Dockerfile
index 717a36c21f..ea5eb74efb 100644
--- a/container/bitnami/Dockerfile
+++ b/container/bitnami/Dockerfile
@@ -1,21 +1,25 @@
 # Copyright Broadcom, Inc. All Rights Reserved.
 # SPDX-License-Identifier: APACHE-2.0
 
+FROM docker.io/bitnami/minideb:bookworm as extractor
+
+COPY automq-*.tgz /tmp/
+RUN mkdir -p /output && \
+    tar -zxf /tmp/automq-*.tgz -C /output --strip-components=1
+
 FROM docker.io/bitnami/minideb:bookworm
 
 ARG DOWNLOADS_URL="downloads.bitnami.com/files/stacksmith"
 ARG JAVA_EXTRA_SECURITY_DIR="/bitnami/java/extra-security"
 ARG TARGETARCH
 
-LABEL com.vmware.cp.artifact.flavor="sha256:c50c90cfd9d12b445b011e6ad529f1ad3daea45c26d20b00732fae3cd71f6a83" \
-      org.opencontainers.image.base.name="docker.io/bitnami/minideb:bookworm" \
-      org.opencontainers.image.created="2025-03-15T19:51:22Z" \
-      org.opencontainers.image.description="Application packaged by Broadcom, Inc." \
-      org.opencontainers.image.documentation="https://github.com/bitnami/containers/tree/main/bitnami/kafka/README.md" \
+LABEL org.opencontainers.image.base.name="docker.io/bitnami/minideb:bookworm" \
+      org.opencontainers.image.created="2025-04-27T21:51:40Z" \
+      org.opencontainers.image.description="AutoMQ packaged by AutoMQ, Inc." \
       org.opencontainers.image.ref.name="3.9.0-debian-12-r13" \
-      org.opencontainers.image.source="https://github.com/bitnami/containers/tree/main/bitnami/kafka" \
-      org.opencontainers.image.title="kafka" \
-      org.opencontainers.image.vendor="Broadcom, Inc." \
+      org.opencontainers.image.source="https://github.com/AutoMQ/automq/tree/main/container" \
+      org.opencontainers.image.title="automq" \
+      org.opencontainers.image.vendor="AutoMQ, Inc." \
       org.opencontainers.image.version="3.9.0"
 
 ENV HOME="/" \
@@ -26,12 +30,11 @@ ENV HOME="/" \
 COPY prebuildfs /
 SHELL ["/bin/bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-c"]
 # Install required system packages and dependencies
-RUN install_packages ca-certificates curl procps zlib1g
-RUN mkdir -p /tmp/bitnami/pkg/cache/ ; cd /tmp/bitnami/pkg/cache/ ; \
+RUN apt-get update && apt-get install -y ca-certificates curl procps zlib1g libjemalloc-dev && \
+    apt-get clean && rm -rf /var/lib/apt/lists /var/cache/apt/archives
+RUN mkdir -p /tmp/bitnami/pkg/cache/ ; cd /tmp/bitnami/pkg/cache/ || exit 1 ; \
     COMPONENTS=( \
-      "wait-for-port-1.0.8-14-linux-${OS_ARCH}-debian-12" \
       "jre-17.0.14-10-1-linux-${OS_ARCH}-debian-12" \
-      "kafka-3.9.0-2-linux-${OS_ARCH}-debian-12" \
     ) ; \
     for COMPONENT in "${COMPONENTS[@]}"; do \
       if [ ! -f "${COMPONENT}.tar.gz" ]; then \
@@ -39,17 +42,18 @@ RUN mkdir -p /tmp/bitnami/pkg/cache/ ; cd /tmp/bitnami/pkg/cache/ ; \
         curl -SsLf "https://${DOWNLOADS_URL}/${COMPONENT}.tar.gz.sha256" -O ; \
       fi ; \
       sha256sum -c "${COMPONENT}.tar.gz.sha256" ; \
-      tar -zxf "${COMPONENT}.tar.gz" -C /opt/bitnami --strip-components=2 --no-same-owner --wildcards '*/files' ; \
+      tar -zxf "${COMPONENT}.tar.gz" -C /opt/bitnami --strip-components=2 --no-same-owner ; \
       rm -rf "${COMPONENT}".tar.gz{,.sha256} ; \
     done
-RUN apt-get update && apt-get upgrade -y && \
-    apt-get clean && rm -rf /var/lib/apt/lists /var/cache/apt/archives
+COPY --from=extractor --chown=1001:0 /output /opt/bitnami/kafka
 RUN chmod g+rwX /opt/bitnami
 RUN find / -perm /6000 -type f -exec chmod a-s {} \; || true
 RUN ln -s /opt/bitnami/scripts/kafka/entrypoint.sh /entrypoint.sh
 RUN ln -s /opt/bitnami/scripts/kafka/run.sh /run.sh
 
 COPY rootfs /
+RUN find /opt/bitnami/scripts -type f -exec chmod g+rwX {} \; && \
+    find /opt/bitnami/scripts -type f -exec chmod +x {} \;
 RUN /opt/bitnami/scripts/java/postunpack.sh
 RUN /opt/bitnami/scripts/kafka/postunpack.sh
 ENV APP_VERSION="3.9.0" \
@@ -59,6 +63,12 @@ ENV APP_VERSION="3.9.0" \
 
 EXPOSE 9092
 
+RUN arch=$(uname -m) \
+    && target_file="/usr/lib/${arch}-linux-gnu/libjemalloc.so" \
+    && { test -f "$target_file" || { echo "Error: $target_file not found"; exit 1; }; } \
+    && ln -sv "$target_file" /usr/lib/libjemalloc.so
+ENV LD_PRELOAD="/usr/lib/libjemalloc.so"
+
 USER 1001
 ENTRYPOINT [ "/opt/bitnami/scripts/kafka/entrypoint.sh" ]
 CMD [ "/opt/bitnami/scripts/kafka/run.sh" ]
diff --git a/container/bitnami/prebuildfs/opt/bitnami/scripts/libbitnami.sh b/container/bitnami/prebuildfs/opt/bitnami/scripts/libbitnami.sh
index 00d053b521..09e3d3084d 100644
--- a/container/bitnami/prebuildfs/opt/bitnami/scripts/libbitnami.sh
+++ b/container/bitnami/prebuildfs/opt/bitnami/scripts/libbitnami.sh
@@ -42,12 +42,13 @@ print_welcome_page() {
 #   None
 #########################
 print_image_welcome_page() {
-    local github_url="https://github.com/bitnami/containers"
+    local docs_url="https://www.automq.com/docs/automq/deployment/deploy-multi-nodes-cluster-on-kubernetes"
 
     info ""
-    info "${BOLD}Welcome to the Bitnami ${BITNAMI_APP_NAME} container${RESET}"
-    info "Subscribe to project updates by watching ${BOLD}${github_url}${RESET}"
-    info "Did you know there are enterprise versions of the Bitnami catalog? For enhanced secure software supply chain features, unlimited pulls from Docker, LTS support, or application customization, see Bitnami Premium or Tanzu Application Catalog. See https://www.arrow.com/globalecs/na/vendors/bitnami/ for more information."
+    info "${BOLD}Welcome to the AutoMQ for Apache Kafka on Bitnami Container${RESET}"
+    info "${BOLD}This image is compatible with Bitnami's container standards.${RESET}"
+    info "Refer to the documentation for complete configuration and Kubernetes deployment guidelines:"
+    info "${BOLD}${docs_url}${RESET}"
     info ""
 }
 
diff --git a/container/bitnami/rootfs/opt/bitnami/scripts/kafka/postunpack.sh b/container/bitnami/rootfs/opt/bitnami/scripts/kafka/postunpack.sh
index 7255563236..673c84e721 100644
--- a/container/bitnami/rootfs/opt/bitnami/scripts/kafka/postunpack.sh
+++ b/container/bitnami/rootfs/opt/bitnami/scripts/kafka/postunpack.sh
@@ -35,12 +35,12 @@ mv "${KAFKA_CONF_DIR}/server.properties" "${KAFKA_CONF_DIR}/server.properties.or
 
 # Disable logging to stdout and garbage collection
 # Source: https://logging.apache.org/log4j/log4j-2.4/manual/appenders.html
-replace_in_file "${KAFKA_BASE_DIR}/bin/kafka-server-start.sh" " [-]loggc" " "
-replace_in_file "${KAFKA_CONF_DIR}/log4j.properties" "DailyRollingFileAppender" "ConsoleAppender"
+#replace_in_file "${KAFKA_BASE_DIR}/bin/kafka-server-start.sh" " [-]loggc" " "
+#replace_in_file "${KAFKA_CONF_DIR}/log4j.properties" "DailyRollingFileAppender" "ConsoleAppender"
 
 # Disable the default console logger in favour of KafkaAppender (which provides the exact output)
-echo "log4j.appender.stdout.Threshold=OFF" >>"${KAFKA_CONF_DIR}/log4j.properties"
+#echo "log4j.appender.stdout.Threshold=OFF" >>"${KAFKA_CONF_DIR}/log4j.properties"
 
 # Remove invalid parameters for ConsoleAppender
-remove_in_file "${KAFKA_CONF_DIR}/log4j.properties" "DatePattern"
-remove_in_file "${KAFKA_CONF_DIR}/log4j.properties" "Appender.File"
+#remove_in_file "${KAFKA_CONF_DIR}/log4j.properties" "DatePattern"
+#remove_in_file "${KAFKA_CONF_DIR}/log4j.properties" "Appender.File"
